### PR TITLE
Feature/modul 318 ajout props et mods visuel dans m dialog

### PR DESCRIPTION
**Pull request by fredduguay**
## `@ulaval/modul-components`
# PR Checklist

- [x] Provide a small description of the changes introduced by this PR
Add multiple new props and behaviors to the dialog component (formerly called modal)

- New dialog types inheriting their visual behavior from the m-message component (similar to State), visual is toggled via prop.
- New mandatory secondary button added.
- Cancel Link is now optional and replaced by the secondary button but can be added after buttons via a prop if need be.
- New optional hint inheriting it's behavior from the m-message component (new state: hint)
- Allow optional primary and secondary/precision labels to be specified for buttons.
- Rename onclick events for the buttons and cancelLink, change emitted value when events are called.
- [x] Include links to issues
https://jira.dti.ulaval.ca/browse/MODUL-318
- [ ] Openshift deployment requested

- [x] Include this section in the release notes
BREAKING CHANGE, multiple modifications to this component, please see below.
----
Add multiple new props and behaviors to the dialog component (formerly called modal)

- New dialog types inheriting their visual behavior from the m-message component (similar to State), visual is toggled via prop.
- New mandatory secondary button added.
- Cancel Link is now optional and replaced by the secondary button but can be added after buttons via a prop if need be.
- New optional hint inheriting it's behavior from the m-message component (new state: hint)
- Allow optional primary and secondary/precision labels to be specified for buttons.
- Rename onclick events for the buttons and cancelLink, change emitted value when events are called.
- [ ] Other info

### CODE REVIEW

File: src/components/dialog/dialog.ts [hunk 1]
```diff
@@ -26,13 +37,17 @@ export class MDialog extends ModulVue implements PortalMixinImpl {
...
     @Prop()
+    public hint: string;
+    @Prop()
     public okLabel: string | undefined;
     @Prop()
     public okPrecision: string | undefined;
     @Prop()
     public cancelLabel: string | undefined;
-    @Prop({ default: true })
-    public negativeLink: boolean;
```
Comment (chuckmah) [inline src/components/dialog/dialog.ts]: Ceci a va causé des breaking change , faudrait le documenter dans la PR

File: src/components/dialog/dialog.ts [hunk 2]
```diff
@@ -57,12 +83,17 @@ export class MDialog extends ModulVue implements PortalMixinImpl {
         return this.$refs.article as HTMLElement;
     }
 
-    private onOk(): void {
+    private onPrimaryButton(): void {
         this.as<PortalMixin>().propOpen = false;
-        this.$emit('ok');
+        this.$emit('primary');
     }
```
Comment (chuckmah) [inline src/components/dialog/dialog.ts]: Ceci a va causé des breaking change , faudrait le documenter dans la PR

File: src/components/dialog/dialog.sandbox.html [hunk 3]
```diff
@@ -5,8 +5,20 @@ <h2 class="m-u--no-margin">Width default</h2>
...
+        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Officiis, delectus, quisquam nobis recusandae, vel doloribus iure sapiente debitis nisi pariatur impedit asperiores magni rerum ratione qui voluptatum dolorum vero id!
+    </m-dialog>
+
+    <h2>Width large with a specified type</h2>
+    <m-dialog class="m-u--margin-top" title="Titre de la fenêtre" type="information" width="large">
+        <m-button slot="trigger">Ouvrir la fenêtre</m-button>
+        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Officiis, delectus, quisquam nobis recusandae, vel doloribus iure sapiente debitis nisi pariatur impedit asperiores magni rerum ratione qui voluptatum dolorum vero id!
+    </m-dialog>
+
+    <h2>Width large with with specified type, hint and negative link</h2>
```
Comment (chuckmah) [inline src/components/dialog/dialog.sandbox.html]: Faudrait changer le titre du test negative link n'existe plus non?

File: src/components/dialog/dialog.scss [hunk 4]
```diff
@@ -112,19 +112,33 @@ $m-dialog--max-width--l: 500px !default;
             padding: $m-spacing $m-spacing;
 
             @media (min-width: $m-mq--min--s) {
-                padding: $m-spacing $m-spacing--m;
+                padding: $m-spacing--l $m-spacing--l;
             }
```
Comment (chuckmah) [inline src/components/dialog/dialog.scss]: on pourrait raccoursir pour 

`padding: $m-spacing--l; `

non?

File: src/components/dialog/dialog.scss [hunk 5]
```diff
@@ -112,19 +112,33 @@ $m-dialog--max-width--l: 500px !default;
...
 
     &__body {
         text-align: center;
+        font-size: $m-font-size--h5;
+        font-weight: $m-font-weight--bold;
 
         @include m-scrollbar();
 
         ~ .m-dialog__footer {
             padding-top: 0;
```
Comment (chuckmah) [inline src/components/dialog/dialog.scss]: Pour mon infom , est-ce qu'il y a une raison de ne pas le mettre directement dans le selecteur  &__footer { au lieu d'utiliser le selecteur "~" ?

File: src/components/message/message.scss [hunk 6]
```diff
@@ -33,6 +33,23 @@ $m-message-size: 32px;
         @include m-message-skin-default($m-color--error);
     }
 
+    &.m--is-state-hint {
+        @include m-message-skin-default($m-color--hint);
+
+        border: 1px solid #00c77f;
+
```
Comment (chuckmah) [inline src/components/message/message.scss]: pas de variable pour cette couleur?

Comment (jipigi) [reply src/components/message/message.scss]: Oui c'est $m-color--success

File: src/components/dialog/dialog.ts (hunk 1 above)
Comment (fredduguay) [reply src/components/dialog/dialog.ts]: Tu as raison, j'en fait l'ajout

File: src/components/dialog/dialog.scss (hunk 5 above)
Comment (fredduguay) [reply src/components/dialog/dialog.scss]: Je n'ai pas fait ce changement, c'est pertinent comme question, je vais vérifier

File: src/components/dialog/dialog.scss (hunk 4 above)
Comment (fredduguay) [reply src/components/dialog/dialog.scss]: yup.

File: src/components/dialog/dialog.ts (hunk 2 above)
Comment (fredduguay) [reply src/components/dialog/dialog.ts]: Je viens d'ajouter, merci.

**Comment by fredduguay** [comment]
@jipigi, à ta demande, je renomme cette remote branch et je vais créer un nouveau PR

**Pull request closed without merging**